test/compiler/datastructures.jl

```diff
@@ -53,3 +53,15 @@ end
         end
     end
 end
+
+# Make sure that the compiler can sort things.
+# https://github.com/JuliaLang/julia/issues/47065
+@testset "Compiler Sorting" begin
```

Comment:
I think this is redundant with (or at least should be in the same place as) test/compiler/sort.jl